### ISSUE TITLE
How to deploy and use eval-only instances?

### ISSUE THREAD

**Issue by a-type** [checkr/flagr#279]
## Expected Behavior

I've deployed a PostgreSQL database to OpenShift, as well as two services with one pod each. 

The first service is `flagr-admin`, which is running the latest `flagr` Docker image and has been configured to talk to Postgres. It starts successfully.

The second is `flagr-eval`, which I am trying to get working with the rest of the system.

I expect `flagr-eval` to act as the "public" part of the feature flagging system; it will serve requests to my clients when they query which feature flags are enabled from browsers, etc. Whereas `flagr-admin` should be kept private, accessible only through our internal network to administrators. Are these expectations correct?

I therefore expect `flagr-eval` to connect in some way to either Postgres, or `flagr-admin`, in order to retrieve and cache the feature flags. The docs mention only two connection types are supported, but don't really elaborate. So I configured it to the `json_http` mode and pointed its "database" connection at the internal cluster address for `flagr-admin`, to see what happens.

## Current Behavior

The admin pod starts up successfully, sets up the database, and announces its availability.

The eval-only pod crashes:

```
  | panic: Get http://<my_flagr_k8s_service_name>.svc.cluster.local:  dial tcp <the_correct_internal_cluster_ip>:80: connect: no route to host
-- | --
  |  
  | goroutine 1 [running]:
  | github.com/checkr/flagr/pkg/handler.(*EvalCache).Start(0xc00063e100)
  | /go/src/github.com/checkr/flagr/pkg/handler/eval_cache.go:50 +0x7a
  | github.com/checkr/flagr/pkg/handler.setupEvaluation(0xc0005ce580)
  | /go/src/github.com/checkr/flagr/pkg/handler/handler.go:73 +0x32
  | github.com/checkr/flagr/pkg/handler.Setup(0xc0005ce580)
  | /go/src/github.com/checkr/flagr/pkg/handler/handler.go:25 +0xea
  | github.com/checkr/flagr/swagger_gen/restapi.configureAPI(0xc0005ce580, 0xc0000321b0, 0x0)
  | /go/src/github.com/checkr/flagr/swagger_gen/restapi/configure_flagr.go:34 +0xbd
  | github.com/checkr/flagr/swagger_gen/restapi.(*Server).ConfigureAPI(...)
  | /go/src/github.com/checkr/flagr/swagger_gen/restapi/server.go:58
  | main.main()
  | /go/src/github.com/checkr/flagr/swagger_gen/cmd/flagr-server/main.go:51 +0x46e
```

Although it resolves the internal IP correctly, it can't connect. Perhaps due to the port?

Before I went any further, I wanted to see if I'm actually trying to do the right thing, here. Should I be trying to connect the eval-only node to the admin (full CRUD) one?

## Steps to Reproduce (for bugs)

Reproduction won't be simple, since I'm using a lot of company custom tooling to perform the deployment. But I believe my question is more of a general one about the intended deployment of the dependent services.

## Context

I'm trying to figure out the correct basic setup for Flagr to test it out in a deployed environment in a similar way to how we might use it in production.

## Your Environment

* Version used (`flagr version`): 1.1.4 (Docker tag)
* Server type and version: OpenShift (Kubernetes) v3.11.98 (k8s version v1.11.0)
* Operating System and version (`uname -a`): I'm using the official Docker container
* Link to your project:

**Comment by a-type** [comment]
Update on this. I'm still not sure about the correct configuration, but I have fixed the first issue I encountered, and I have a theory on what to do next.

The `no route to host` error was due to my Kubernetes Service for the flagr management instance not routing traffic because the pod was not passing a health check. I put in the wrong health check endpoint (it should be `/api/v1/health`). Once I fixed that, the Service initialized correctly and began routing traffic to the pod, and the eval-only instance could then talk to the manager via its Service URL.

Now I just have to figure out what the Eval service should be doing. I have tentatively updated its connection string to point directly to the JSON export API endpoint of the manager service. I'll see if that seems to work or not...
